### PR TITLE
Add validation to patchinfo in the Rails way

### PR DESCRIPTION
**Pull request by saraycp**
As Patchinfo is not an ApplicationRecord, we use ActiveModel::Model to
use Rails validators and set the required attr_accessors.

It also refactors patchinfo initialize method, passing data as argument.

Co-authored-by: David Kang <dkang@suse.com>
Co-authored-by: Eduardo Navarro <enavarro@suse.com>

### CODE REVIEW

**Comment by saraycp** [comment]
[Refactor patchinfo controller](https://trello.com/c/4JGl1Kkx/1094-refactor-patchinfo-controller)

File: src/api/app/models/patchinfo.rb [hunk 1]
```diff
@@ -37,14 +38,23 @@ class TrackerNotFound < APIError
   CATEGORIES = (CATEGORY_COLORS.keys << '').freeze
 
   attr_reader :document
+  attr_writer :data
+  attr_accessor :summary, :description, :packager
+
+  validates :summary, length: { minimum: 11, message: 'is too short (should have more than 10 signs)' }
```
Comment (hennevogel) [inline src/api/app/models/patchinfo.rb]: s/signs/characters/

File: src/api/app/models/patchinfo.rb [hunk 2]
```diff
@@ -247,4 +257,10 @@ def issues_by_tracker
     end
     issues_by_tracker
   end
+
+  def description_length
+    unless description && description.length > [summary.length, 50].max
```
Comment (hennevogel) [inline src/api/app/models/patchinfo.rb]: I find this hard to understand and doubt it makes sense.

**Comment by hennevogel** [comment]
Please also look into input [patterns](https://www.w3.org/TR/2009/WD-html5-20090825/forms.html#attr-input-pattern) for the UI...

File: src/api/app/models/patchinfo.rb (hunk 1 above)
Comment (saraycp) [reply src/api/app/models/patchinfo.rb]: done

File: src/api/app/models/patchinfo.rb (hunk 2 above)
Comment (saraycp) [reply src/api/app/models/patchinfo.rb]: @hennevogel Now we don't compare description length with summary length, only check if description is bigger than 50. Better now?

**Comment by saraycp** [comment]
@hennevogel we have used `minlength` attribute instead of `pattern`. 

We also plan to keep the current views like they are and create new views and the controller for bootstrap.

**Pull request merged**